sdk/wallet-adapter/packages/adapters/wallet-standard-adapter/src/StandardWalletAdapter.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { SignableTransaction } from "@mysten/sui.js";
+import { WalletAdapter } from "@mysten/wallet-adapter-base";
+import { StandardWalletAdapterWallet } from "@mysten/wallet-standard";
+
+export interface StandardWalletAdapterConfig {
+  wallet: StandardWalletAdapterWallet;
+}
+
+export class StandardWalletAdapter implements WalletAdapter {
+  connected = false;
+  connecting = false;
+
+  #wallet: StandardWalletAdapterWallet;
+
+  constructor({ wallet }: StandardWalletAdapterConfig) {
+    this.#wallet = wallet;
+  }
+
+  get name() {
+    return this.#wallet.name;
+  }
+
+  get icon() {
+    return this.#wallet.icon;
+  }
+
+  get wallet() {
+    return this.#wallet;
+  }
+
+  async getAccounts() {
+    return this.#wallet.accounts.map((account) => account.address);
+  }
+
+  async connect() {
+    try {
+      if (this.connected || this.connecting) return;
+      this.connecting = true;
+
+      if (!this.#wallet.accounts.length) {
+        await this.#wallet.features["standard:connect"].connect();
+      }
+
+      if (!this.#wallet.accounts.length) {
+        throw new Error("No wallet accounts found");
+      }
+
+      this.connected = true;
+    } finally {
+      this.connecting = false;
+    }
+  }
+
+  async disconnect() {
+    this.connected = false;
+    this.connecting = false;
+  }
```

Comment:
Wallet's diconnect method is also required! Wallet may need a trigger for its cleanup logic after user disconnnects it.

Reply:
What kind of cleanup logic are you imagining? Ideally wallet connections to should be stateless.

Reply:
> What kind of cleanup logic are you imagining? Ideally wallet connections to should be stateless.

For example, when Suiet wallet receives the `connect` call from `StandardWalletAdapter`, it would setup a window message listener; So we need `StandardWalletAdapter` to call our `disconnect` function to cleanup the listener.

What I expected be like:

```
  async disconnect() {
    await wallet.disconnect();
    this.connected = false;
    this.connecting = false;
  }
```

Reply:
I'll talk to the Solana folks to get their input here.

Reply:
> I'll talk to the Solana folks to get their input here.

So nice. Thank you

Reply:
Wallets generally attach the message listener before a connection request is initiated, is there a reason why you're doing it lazily?

Reply:
Well for this point, at first I try to avoid setting up useless listeners because not all the pages are dapps (We did inject content script to every page by wildcard url). But now I find out this might not be a good practice cuz an idle service worker would get killed by Chrome 😂

Reply:
If you only allow communication through the wallet standard, it would still satisfy this requirement. When the wallet interface is actually registered (when we iterate through the `navigator.wallets`), the listeners would get attached, which would only happen on Sui dapps.

We added a disconnect feature as well just for completeness.